app/src/main/java/org/mozilla/reference/browser/addons/WebExtensionPromptFeature.kt

```diff
@@ -0,0 +1,155 @@
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.reference.browser.addons
+
+import android.content.Context
+import android.view.View
+import android.widget.Toast
+import androidx.annotation.VisibleForTesting
+import androidx.fragment.app.FragmentManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.action.WebExtensionAction
+import mozilla.components.browser.state.state.extension.WebExtensionPromptRequest
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.feature.addons.Addon
+import mozilla.components.feature.addons.ui.PermissionsDialogFragment
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import org.mozilla.reference.browser.R
+
+/**
+ * Feature implementation for handling [WebExtensionPromptRequest] and showing the respective UI.
+ */
+class WebExtensionPromptFeature(
```

Comment:
Should this be upstreamed to AC?

Reply:
Yes, we could file a follow up ticket :)

Reply:
This way we can unblock the AC update.

Reply:
Now that I'm working on https://bugzilla.mozilla.org/show_bug.cgi?id=1825124, I'm starting to notice that porting this feature to AC may not be worthwhile as it contains at lot of specific UI about how browsers handle extension prompts that I think the parts are shared a minuscule vs all the heavy UI work to we will have to generalized to have a shared component.